text/0000-export-run-image-metadata.md

```diff
@@ -0,0 +1,139 @@
+
+Putting this data into `report.toml` is advantageous over other methods, especially when considering the kubernetes `terminationMessagePath` message [pattern](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#pod-v1-core). In this pattern, the content of `report.toml` can be used as a container's termination message, making this data easily accessible after an image is exported or rebased within a kubernetes container.
+
+# What it is
+[what-it-is]: #what-it-is
+
+These values will be included when the lifecycle exporter/creator/rebaser binary writes `report.toml`.
+
+Here are two examples of `report.toml` content. (Other values are omitted for readability.)
+
+#### Image published to a registry:
+```
+[image]
+tags = ...
+digest = ...
+image-id = ...
+manifest-size = ...
+run-image.image = "run/name:foo"
+run-image.reference = "index.docker.io/run/name@sha256:94f85561b0976bf1e2bef6b14de92299ebcd4c8148802cf9b217654651e4f416"
+run-image.top-layer = "sha256:83ad2f0b091621ce19357e19d853c8be1b8f4d60d99c281fc2db75e0f56df42a"
+run-image.mirrors = ["<mirror1>", "<mirror2>"]
+```
+
+#### Image exported to the docker daemon:
+```
+[image]
+tags = ...
+digest = ...
+image-id = ...
+manifest-size = ...
+run-image.image = "run/name:foo"
+run-image.reference = "5b90f9c0e189"
+run-image.top-layer = "sha256:83ad2f0b091621ce19357e19d853c8be1b8f4d60d99c281fc2db75e0f56df42a"
+run-image.mirrors = ["<mirror1>", "<mirror2>"]
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+This metadata is readily available when `report.toml` is created, so it will be straight-forward to extend `report.toml`.
+
+# Migration
+[migration]: #migration
+
+N/A
+
+This is an additive change to a metadata file and will be backwards compatible.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This metadata is written to the [`lifecycle.metadata` label](https://github.com/buildpacks/spec/blob/main/platform.md#iobuildpackslifecyclemetadata-json) and it can be accessed by querying a docker daemon or registry. So we will be writing this data to two outputs.
+
+# Alternatives
+[alternatives]: #alternatives
+
+1. Do nothing and continue to require platforms to retrieve this information via alternative means--either querying the docker daemon or registry.
+    - Rebase process may remain suboptimal for some platform providers.
+
+2. Write all the metadata labels to `report.toml`.
+    - This could break platform operators that are using the `terminationMessagePath` [pattern](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#pod-v1-core). Because some of the metadata includes unbounded arrays, we could explode this report beyond the max size of 4096 bytes.
```

Comment:
@tomkennedy513 do you have any feedback here?

Reply:
FWIW I don't think we're close to hitting the limit yet (though it's been said before...)

Reply:
IIRC kpack doesn't blindly dump the report.toml directly, it parses out [only the stuff it cares about](https://github.com/buildpacks-community/kpack/blob/ae5eb15b8b9632fda50bafc4d2fad0f8c2c8a2a8/pkg/cnb/build_metadata.go#L22-L28) and [writes that to `terminationMessagePath` instead](https://github.com/buildpacks-community/kpack/blob/ae5eb15b8b9632fda50bafc4d2fad0f8c2c8a2a8/pkg/cnb/build_metadata.go#L22-L28).

I don't think we've encountered any issues with length yet, but I image that's because we limit the unbounded array to just the id, version, and homepage metadata of buildpacks.